packages/contracts/src/2.0.0/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -90,6 +90,12 @@ contract MixinExchangeCore is
     {
         // Fetch order info
         OrderInfo memory orderInfo = getOrderInfo(order);
+        
+        // An order can only be filled if its status is FILLABLE.
+        require(
```

Comment:
It is, but if we wait until the call to `assertValidFill` it will fail on a math error in `calculateFillResults`.

If we check it here (though redundantly) we return the correct error message.